libraries/engage/core/hooks/useWidgetSettings.js

```diff
@@ -1,15 +1,15 @@
-import { useWidgetConfig } from './useWidgetConfig';
+import { useRoute } from './useRoute';
+import { getWidgetSettings } from '../config/getWidgetSettings';
 
 /**
- * Retrieves the widget-configuration-settings for a specific widget by its id and
- * an optional index.
- * If no index is given it will return the first found widget-configuration-settings by widgetId.
+ * Retrieves inherited settings for a widget given by its id and optional index on the current page.
+ * See documentation of `getWidgetSettings` for further details.
  *
  * @param {string} widgetId The id of the widget to look for, which must exist in the config.
- * @param {number|undefined} [index] The optional index of the widget.
+ * @param {number|undefined} [index=0] The optional index of the widget.
  * @returns {Object}
  */
-export function useWidgetSettings(widgetId, index) {
-  const { settings = {} } = useWidgetConfig(widgetId, index);
-  return settings;
+export function useWidgetSettings(widgetId, index = 0) {
+  const { pattern: pagePattern } = useRoute();
```

Comment:
This is not needed if the `useCurrentRoute()` helper is used inside `getPageConfig()` as mentioned above.

Reply:
See my previous comment.